### PR TITLE
Update k8s status command

### PR DESCRIPTION
**Pull request by bschimke95**
Example output:
```
ubuntu@alice:~/k8s$ sudo k8s status
status: ready
high-availability: no
datastore:
  master-nodes:
    - 10.84.220.237:6400
  standy-nodes: none
  spare-nodes:
    - 10.84.220.25:6400

network:
  enabled: true
dns:
  enabled: true
  cluster-domain: cluster.local
  service-ip: 10.152.183.108
  upstream-nameservers:
  - /etc/resolv.conf
metrics-server:
  enabled: true
```

### CODE REVIEW

File: src/k8s/cmd/k8s/k8s_bootstrap.go [hunk 1]
```diff
@@ -41,21 +41,23 @@ func newBootstrapCmd() *cobra.Command {
...
 				config.SetDefaults()
 			}
+			const minTimeout = 3 * time.Second
+			if bootstrapCmdOpts.timeout < minTimeout {
+				cmd.PrintErrf("Timeout %v is less than minimum of %v. Using the minimum %v instead.\n", bootstrapCmdOpts.timeout, minTimeout, minTimeout)
+				bootstrapCmdOpts.timeout = minTimeout
+			}
+
+			timeoutCtx, cancel := context.WithTimeout(cmd.Context(), joinNodeCmdOpts.timeout)
+			defer cancel()
```
Comment (bschimke95) [inline src/k8s/cmd/k8s/k8s_bootstrap.go:57]: drive-by fix: context was not using the timeout

File: src/k8s/api/v1/types.go [hunk 2]
```diff
@@ -119,20 +119,74 @@ func (c ClusterStatus) String() string {
...
+	if len(voters) > 0 {
+		result.WriteString(fmt.Sprintf("  master-nodes:\n"))
+		for _, voter := range voters {
+			result.WriteString(fmt.Sprintf("    - %s\n", voter.Address))
+		}
+	} else {
+		result.WriteString(fmt.Sprintf("  master-nodes: none\n"))
+	}
+	if len(standBys) > 0 {
+		result.WriteString(fmt.Sprintf("  standy-nodes:\n"))
```
Comment (eaudetcobello) [inline src/k8s/api/v1/types.go:147]:
```suggestion
		result.WriteString(fmt.Sprintf("  standby-nodes:\n"))
```

File: src/k8s/api/v1/types.go [hunk 3]
```diff
@@ -119,20 +119,74 @@ func (c ClusterStatus) String() string {
...
+	} else {
+		result.WriteString(fmt.Sprintf("  standy-nodes: none\n"))
+	}
+	if len(spares) > 0 {
+		result.WriteString(fmt.Sprintf("  spare-nodes:\n"))
+		for _, spare := range spares {
+			result.WriteString(fmt.Sprintf("    - %s\n", spare.Address))
+		}
+	} else {
+		result.WriteString(fmt.Sprintf("  spare-nodes: none\n"))
```
Comment (eaudetcobello) [inline src/k8s/api/v1/types.go:160]: Could a function be added to reduce repetitiveness of building the string?

File: src/k8s/api/v1/types.go [hunk 4]
```diff
@@ -119,20 +119,74 @@ func (c ClusterStatus) String() string {
...
+		printedConfig.Ingress = c.Config.Ingress
+	}
+	if c.Config.LoadBalancer.Enabled != nil && *c.Config.LoadBalancer.Enabled {
+		printedConfig.LoadBalancer = c.Config.LoadBalancer
 	}
+	if c.Config.LocalStorage.Enabled != nil && *c.Config.LocalStorage.Enabled {
+		printedConfig.LocalStorage = c.Config.LocalStorage
+	}
+	if c.Config.Gateway.Enabled != nil && *c.Config.Gateway.Enabled {
+		printedConfig.Gateway = c.Config.Gateway
```
Comment (eaudetcobello) [inline src/k8s/api/v1/types.go:181]: How about something like this?

```suggestion
		updatePrintedConfigIfEnabled(c.Config.Network, func() {
             printedConfig.Network = c.Config.Network
        })
        updatePrintedConfigIfEnabled(c.Config.DNS, func() {
             printedConfig.DNS = c.Config.DNS
        })
```

**Review by eaudetcobello — COMMENTED** [review]
Left some comments :)

File: src/k8s/api/v1/types.go (hunk 4 above)
Comment (bschimke95) [reply src/k8s/api/v1/types.go:181]: hm, the `updatePrintedConfigIfEnabled` would just abstract away the nil check, right?
I don't think this is more readable or shorter than the if statement, is it?

File: src/k8s/api/v1/types.go (hunk 3 above)
Comment (bschimke95) [reply src/k8s/api/v1/types.go:160]: Hm, I think it may be slightly more readable but not much.

Also, I would put such function into the `utils` package but we cannot do this as this would cause cyclic dependencies.

A bit of repetitiveness is the price we pay with the explicit configuration, I guess

File: src/k8s/api/v1/types.go (hunk 4 above)
Comment (eaudetcobello) [reply src/k8s/api/v1/types.go:181]: That is true

File: src/k8s/api/v1/types.go [hunk 5]
```diff
@@ -118,20 +119,74 @@ func (c ClusterStatus) String() string {
...
+		case DatastoreRoleVoter:
+			voters = append(voters, node)
+		case DatastoreRoleStandBy:
+			standBys = append(standBys, node)
+		case DatastoreRoleSpare:
+			spares = append(spares, node)
+		}
+	}
+	if len(voters) > 0 {
+		result.WriteString(fmt.Sprintf("  master-nodes:\n"))
```
Comment (ktsakalozos) [inline src/k8s/api/v1/types.go:139]: Do not use the word "master", please use "voter"

**Comment by bschimke95** [comment]
Good point - removed `omitempty` for configuration values and only left it for the functionalities (that way I can set a functionality to nil and it is not rendered by the k8s status command) 

Output now:
```
ubuntu@bob:~/k8s$ sudo k8s status
status: ready
high-availability: no
datastore:
  voter-nodes:
    - 10.84.220.25:6400
  standy-nodes: none
  spare-nodes: none

dns:
  enabled: true
  cluster-domain: cluster.local
  service-ip: 10.152.183.233
  upstream-nameservers:
  - /etc/resolv.conf
ingress:
  enabled: true
  default-tls-secret: ""
  enable-proxy-protocol: false
metrics-server:
  enabled: true
```

**Review by berkayoz — APPROVED** [review]
LGTM

File: src/k8s/pkg/utils/database.go [hunk 6]
```diff
@@ -28,6 +30,99 @@ func GetClusterConfig(ctx context.Context, state *state.State) (types.ClusterCon
 	return clusterConfig, nil
 }
 
+// GetUserFacingClusterConfig returns the public cluster config.
+func GetUserFacingClusterConfig(ctx context.Context, state *state.State) (apiv1.UserFacingClusterConfig, error) {
```
Comment (neoaggelos) [inline src/k8s/pkg/utils/database.go:34]: An important note for this. This is not `utils`: `utils` are supposed to be stand-alone helpers.

I see why this is here, same as the others in the same file. But we have some them in `api/impl` as well. The problem is that this does not allow importing the `pkg/utils` package from `pkg/k8sd/database` code, which is not ideal.

It's OK to deal with this at a later time.

**Review by neoaggelos — APPROVED** [review]

**Review by louiseschmidtgen — APPROVED** [review]
LGTM
